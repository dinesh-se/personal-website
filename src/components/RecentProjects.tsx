import Link from 'next/link';

import { getRepos } from '@api/github';
import IconCode from '@root/public/assets/code.svg';

const RecentProjects = async () => {
  const params = {
    per_page: '3'
  };
  const repos = await getRepos(params);

  return (
    <>
      <h3 className="flex font-semibold tracking-wider space-x-4 mb-10">
        <IconCode className="h-8 w-8 svg-icon" />
        <span className="text-2xl">Recent Works</span> 
      </h3>
      <div className="flex flex-col space-y-10">
        {repos.map(({node_id: nodeId, name, html_url: htmlUrl, description}) => (
          <article className="group relative flex flex-col" key={nodeId}>
            <h2 className="font-semibold tracking-wider text-zinc-100 mb-3">
            <div className="absolute -inset-y-6 -inset-x-4 z-0 scale-95 bg-neutral-200 dark:bg-zinc-800/50 opacity-0 transition group-hover:scale-100 group-hover:opacity-100 sm:-inset-x-6 sm:rounded-2xl"></div>
              <Link className="relative z-10 text-sky-500 hover:text-sky-600 dark:hover:text-sky-400" href={htmlUrl} target='_blank'>{name}</Link>
            </h2>
            <p className="relative z-10 text-sm text-zinc-500 dark:text-zinc-400">
              {description}
            </p>
          </article>
        ))}
      </div>
    </>
  )
}

export default RecentProjects
