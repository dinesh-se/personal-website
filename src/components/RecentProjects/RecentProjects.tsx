import Link from 'next/link';

import IconCode from '@root/public/assets/code.svg';

import { Projects } from '@types';

const RecentProjects = ({ projects }: Projects) => {
	return (
		<>
			<h3 className="mb-10 flex space-x-4 font-semibold tracking-wider">
				<IconCode className="svg-icon h-8 w-8" />
				<span className="text-2xl">Recent Works</span>
			</h3>
			<div className="flex flex-col space-y-10">
				{projects.map(({ id, name, url, description }) => (
					<article className="group relative flex flex-col" key={id}>
						<h2 className="mb-3 font-semibold tracking-wider text-zinc-100">
							<div className="absolute -inset-x-4 -inset-y-6 z-0 scale-95 bg-neutral-200 opacity-0 transition group-hover:scale-100 group-hover:opacity-100 sm:-inset-x-6 sm:rounded-2xl dark:bg-zinc-800/50"></div>
							<Link
								className="relative z-10 text-sky-500 hover:text-sky-600 dark:hover:text-sky-400"
								href={url}
								target="_blank"
							>
								{name}
							</Link>
						</h2>
						<p className="relative z-10 text-sm text-zinc-500 dark:text-zinc-400">
							{description}
						</p>
					</article>
				))}
			</div>
		</>
	);
};

export default RecentProjects;
