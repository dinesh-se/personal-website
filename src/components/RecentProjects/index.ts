export { default as RecentProjects } from './RecentProjects';
