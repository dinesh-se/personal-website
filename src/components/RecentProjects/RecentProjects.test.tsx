import { render } from '@testing-library/react';

import { Projects } from '@types';

import RecentProjects from './RecentProjects';

describe('RecentProjects', () => {
	it('renders correctly', async () => {
		const props: Projects = {
			projects: [
				{
					id: 'id-1',
					name: 'Repo 1',
					url: 'https://github/repo/1',
					description: 'Github Repository 1',
				},
				{
					id: 'id-2',
					name: 'Repo 2',
					url: 'https://github/repo/2',
					description: 'Github Repository 2',
				},
				{
					id: 'id-3',
					name: 'Repo 3',
					url: 'https://github/repo/3',
					description: 'Github Repository 3',
				},
			],
		};
		const { container } = render(<RecentProjects {...props} />);
		expect(container).toMatchSnapshot();
	});
});
