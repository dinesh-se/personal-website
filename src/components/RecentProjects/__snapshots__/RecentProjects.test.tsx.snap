// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecentProjects renders correctly 1`] = `
<div>
  <h3
    class="mb-10 flex space-x-4 font-semibold tracking-wider"
  >
    <svg
      class="svg-icon h-8 w-8"
    />
    <span
      class="text-2xl"
    >
      Recent Works
    </span>
  </h3>
  <div
    class="flex flex-col space-y-10"
  >
    <article
      class="group relative flex flex-col"
    >
      <h2
        class="mb-3 font-semibold tracking-wider text-zinc-100"
      >
        <div
          class="absolute -inset-x-4 -inset-y-6 z-0 scale-95 bg-neutral-200 opacity-0 transition group-hover:scale-100 group-hover:opacity-100 sm:-inset-x-6 sm:rounded-2xl dark:bg-zinc-800/50"
        />
        <a
          class="relative z-10 text-sky-500 hover:text-sky-600 dark:hover:text-sky-400"
          href="https://github/repo/1"
          target="_blank"
        >
          Repo 1
        </a>
      </h2>
      <p
        class="relative z-10 text-sm text-zinc-500 dark:text-zinc-400"
      >
        Github Repository 1
      </p>
    </article>
    <article
      class="group relative flex flex-col"
    >
      <h2
        class="mb-3 font-semibold tracking-wider text-zinc-100"
      >
        <div
          class="absolute -inset-x-4 -inset-y-6 z-0 scale-95 bg-neutral-200 opacity-0 transition group-hover:scale-100 group-hover:opacity-100 sm:-inset-x-6 sm:rounded-2xl dark:bg-zinc-800/50"
        />
        <a
          class="relative z-10 text-sky-500 hover:text-sky-600 dark:hover:text-sky-400"
          href="https://github/repo/2"
          target="_blank"
        >
          Repo 2
        </a>
      </h2>
      <p
        class="relative z-10 text-sm text-zinc-500 dark:text-zinc-400"
      >
        Github Repository 2
      </p>
    </article>
    <article
      class="group relative flex flex-col"
    >
      <h2
        class="mb-3 font-semibold tracking-wider text-zinc-100"
      >
        <div
          class="absolute -inset-x-4 -inset-y-6 z-0 scale-95 bg-neutral-200 opacity-0 transition group-hover:scale-100 group-hover:opacity-100 sm:-inset-x-6 sm:rounded-2xl dark:bg-zinc-800/50"
        />
        <a
          class="relative z-10 text-sky-500 hover:text-sky-600 dark:hover:text-sky-400"
          href="https://github/repo/3"
          target="_blank"
        >
          Repo 3
        </a>
      </h2>
      <p
        class="relative z-10 text-sm text-zinc-500 dark:text-zinc-400"
      >
        Github Repository 3
      </p>
    </article>
  </div>
</div>
`;
